### DISCUSSION TITLE
CI/CD enhancement

### DISCUSSION THREAD

**Discussion by malcolmSansen** [Sansen-Shimizu/sakura-boot#12] [Ideas]
Discussion related to the [CI/CD issue](https://github.com/Sansen-Shimizu/sakura-boot/issues/8).

**Comment by malcolmSansen** [comment]
@vr-varad do you need any information about the project ?
My first idea for the CI was a simple workflow: static analysis -> build -> check and test -> deploy snapshot. On every commit.
And for the CD same as CI but deploy the version of the new release tag. On tag creation.

For the documentation it's already done but it can be adapt.
The deployment is on maven central and for the snapshot it will use the version on the gradle.properties file.

If I miss anything or if there is any improvement that can be done I will be happy to hear it.

**Comment by vr-varad** [comment]
As of now I will start working on this and as I face any issue I will get it cleared here.

**Reply by malcolmSansen** [reply]
Ok thank you!
